app/models/foreman_tasks/task.rb

```diff
@@ -223,6 +227,15 @@ def to_label
       parts.join(' ').strip
     end
 
+    # Method that produces hash where key is output type (ie. stdout, stderr...) and value is output for given type
+    # So the returned hash look like {'stdout': 'normal output', 'stderr': 'error'}
+    def output_by_type
+      return {} unless main_action && main_action.live_output.any?
```

Comment:
There's only one action class which has `#live_output` defined. If I'm reading this right, this would raise an `NoMethodError` if called on anything else than `Actions::RemoteExecution::RunHostJob`